tracee-ebpf/btfinfo.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"github.com/aquasecurity/tracee/tracee-ebpf/tracee"
+	"os"
+	"strings"
+)
+
+func fileExists(path string) bool {
+	if _, err := os.Stat(path); err == nil {
+		return true
+	}
+	return false
+}
+
+func fileGrepUnique(path string, needle string) string {
+	file, _ := os.Open(path)
+	defer func(file *os.File) {
+		_ = file.Close()
+	}(file)
+	scanner := bufio.NewScanner(file)
+	for i := 1; scanner.Scan(); i++ {
+		if strings.Contains(scanner.Text(), needle) {
+			return scanner.Text()
+		}
+	}
+	return ""
+}
+
+// NewBTFInfo creates a BTFInfo object and runs DiscoverDistro()
+// on its creation. It returns nil if no discovery data was
+// avalable. To get specific error from DiscoverDistro method,
+// user should create a BTFInfo struct and call DiscoverDistro
+// by itself.
+func NewBTFInfo() *BTFInfo {
+	btfi := new(BTFInfo)
+	if err := btfi.DiscoverDistro(); err != nil {
+		return nil
+	}
+	return btfi
+}
+
+// BTFInfo
+
+type BTFInfo struct {
+	distroId     string
+	distroVer    string
+	distroKernel string
+}
+
+func (btfi *BTFInfo) GetDistroId() string {
+	return btfi.distroId
+}
+
+func (btfi *BTFInfo) GetDistroVer() string {
+	return btfi.distroVer
+}
+
+func (btfi *BTFInfo) GetDistroKernel() string {
+	return btfi.distroKernel
+}
+
+// DiscoverDistro tries to discover running Linux distribution either by reading
+// /etc/os-releases (https://man7.org/linux/man-pages/man5/os-release.5.html) or
+// by interpreting the version out of uname(2) system call.
+func (btfi *BTFInfo) DiscoverDistro() error {
+
+	ker := tracee.UnameRelease()
+	if ker == "" {
+		return errors.New("could not get uname")
+	}
+
+	osrelease := "/etc/os-release"
```

Comment:
Do all distros with `/etc/os-release` use this same format? I can confirm for Fedora :-)

Reply:
http://0pointer.de/blog/projects/os-release

This is the announce that says systemd stopped supporting distros that does not implement it AND it is a standard by freedesktop iirc. 

Nevertheless, idea is to either expand discovery function or to have different discovery mechanisms ready if needed. For initial BTF support I think this is enough as we can always expand this later.